Tools/CMake/AMReX_Options.cmake

```diff
@@ -90,15 +90,33 @@ endif ()
 print_option( DIM )
 
 #
-# Programming models  ========================================================
+# Parallel backends    ========================================================
 #
-option( ENABLE_MPI  "Enable MPI"  ON)
+
+# For the time being ENABLE_DPCPP is defined before project() is called
+# Check whether the C++ compiler is dpcpp
+print_option(ENABLE_DPCPP)
+if (ENABLE_DPCPP AND (NOT (CMAKE_CXX_COMPILER MATCHES "dpcpp") ) )
+   message(FATAL_ERROR "\nENABLE_DPCPP=${ENABLE_DPCPP} but CXX compiler is not dpcpp\n")
+endif ()
+
+cmake_dependent_option( ENABLE_DPCPP_AOT  "Enable DPCPP AOT"  OFF
+   "ENABLE_DPCPP" OFF)
+print_option( ENABLE_DPCPP_AOT )
+
+cmake_dependent_option( ENABLE_DPCPP_SPLIT_KERNEL "Enable DPCPP kernel splitting"  ON
+   "ENABLE_DPCPP" OFF)
+print_option(  ENABLE_DPCPP_SPLIT_KERNEL )
+
+cmake_dependent_option( ENABLE_MPI  "Enable MPI"  ON
+   "NOT ENABLE_DPCPP" OFF)
 print_option( ENABLE_MPI )
 
 option( ENABLE_OMP  "Enable OpenMP" OFF)
 print_option( ENABLE_OMP )
 
-option( ENABLE_CUDA "Enable GPU support via CUDA" OFF )
+cmake_dependent_option( ENABLE_CUDA "Enable GPU support via CUDA" OFF
```

Comment:
I think this is more of a use-case for a conflict/fatal error than a dependent option, but am happy to be convinced otherwise with example workflows.

Reply:
As for MPI, In the GNUMake CUDA is turned off if DPC++ is on. From the CMake doc: ```[cmake_dependent_option]  presents an option to the user only if a set of other conditions are true.``` We don't want to preset ENABLE_CUDA as an option if DPC++ is enabled, so I would argue this falls into the cmake_dependent_option usage

Reply:
But how would a user then switch back with `ccmake` from DPC++ to CUDA? Would probably need 2x reconfig (disable DPC++, reconfig, enable CUD, reconfig, generate), which is a bit involved I think.

Reply:
You can set ENABLE_CUDA an any time but it will not be used until you switch off ENABLE_DPCPP.
[https://cmake.org/cmake/help/v3.14/module/CMakeDependentOption.html](https://cmake.org/cmake/help/v3.14/module/CMakeDependentOption.html)